docs/dev/debug.md

```diff
@@ -0,0 +1,267 @@
+operators and their dependencies.  In our case this points to the index image,
+that runs a grpc service as mentioned before.
+
+This step is necessary because our operator we are installing is not present in
+the default catalog included in OpenShift.
+
+At this point OpenShift knows everything about our operators, but we still need
+to tell it that we want to install that specific operator.  This is where the
+[`Subscription`](
+https://olm.operatorframework.io/docs/concepts/crds/subscription/) comes into
+play.
+
+A Subscription represents an intention to install an operator and the `make
+openstack` command creates one for our operator and specifies our
+`CatalogSource` as the source to find our operator.  This means that it will
+install our custom operator even if we have an official operator already
+released in the official operator catalog.
+
+This newly created `Subscription` triggers the creation of the index pod so the
+OLM can query the information, then an [`InstallPlan`](
+https://docs.openshift.com/container-platform/4.11/rest_api/operatorhub_apis/installplan-operators-coreos-com-v1alpha1.html)
+gets created to take care of installing the resources for the operator.
+
+If the operators are not correctly deployed after running `make openstack`,
+then we should look into the `InstallPlan` and check for errors.
+
+```
+oc describe InstallPlan | less
+```
+
+If there is no `InstallPlan` we have to check if the index pod is running:
+
+```
+oc get pod -l olm.catalogSource=openstack-operator-index
+```
+
+If it isn't, check for the `CatalogSource`:
+
+```
+oc describe catalogsource openstack-operator-index
+```
+
+##### Propagating CRs
+
+When we run `make openstack_deploy` we are basically applying our
+`OpenStackControlPlane` manifest, as defined in the `OPENSTACK_CR`
+environmental variable, which is a CRD defined by the openstack-operator.
+
+The openstack-operator has a controller watching for `OpenStackControlPlane`
+resources, so when it sees a new one it starts working to reconcile it. In this
+case that means propagating the `template` in the `glance` section into a new
+`Glance` resource.
+
+So after the `OpenStackControlPlane` resource we should be able to see the
+`Glance` resource created by the openstack-operator, and this should contain
+the same information present in the `template` section in the `glance` section
+of our manifest.
+
+```
+oc get glance
+```
+
+If we don't see this resource, then we need to check first that the `enabled`
+key inside the `glance` section is not set to `false` and then look at the
+openstack-operator logs and search for reconciliation errors.
+
+```
+OPENSTACK_OPERATOR=`oc get pod -l control-plane=controller-manager -o custom-columns=POD:.metadata.name|grep openstack`
+oc logs $OPENSTACK_OPERATOR
+```
+
+Something similar should happen for the keystone and cinder operators.
+
+```
+oc get keystoneapis
+oc get cinder
```

Comment:
s/cinder/glance

Reply:
I think this should be cinder only as it is given as example/

Reply:
ok to keep the example on cinder.